packages/core/src/readers/LlamaParseReader.ts

```diff
@@ -325,49 +368,250 @@ export class LlamaParseReader extends FileReader {
+      try {
+        await fs.access(downloadPath);
+      } catch {
+        await fs.mkdir(downloadPath, { recursive: true });
+      }
 
-          if (!imagePath.endsWith(".png") && !imagePath.endsWith(".jpg")) {
-            imagePath += ".png";
+      const images: Record<string, any>[] = [];
+      for (const result of jsonResult) {
+        const jobId = result.job_id;
+        for (const page of result.pages) {
+          if (this.verbose) {
+            console.log(`> Image for page ${page.page}: ${page.images}`);
           }
+          for (const image of page.images) {
+            const imageName = image.name;
+            // Get the full path
+            let imagePath = `${downloadPath}/${jobId}-${imageName}`;
 
-          // Get a valid image path
-          image.path = imagePath;
-          image.job_id = jobId;
-          image.original_pdf_path = result.file_path;
-          image.page_number = page.page;
-
-          const imageUrl = `${this.baseUrl}/job/${jobId}/result/image/${imageName}`;
-          const response = await fetch(imageUrl, { headers });
-          if (!response.ok) {
-            throw new Error(
-              `Failed to download image: ${await response.text()}`,
-            );
-          }
-          const arrayBuffer = await response.arrayBuffer();
-          const buffer = Buffer.from(arrayBuffer);
-          await fs.writeFile(imagePath, buffer);
+            if (!imagePath.endsWith(".png") && !imagePath.endsWith(".jpg")) {
+              imagePath += ".png";
+            }
+
+            // Get a valid image path
+            image.path = imagePath;
+            image.job_id = jobId;
+            image.original_pdf_path = result.file_path;
+            image.page_number = page.page;
 
-          images.push(image);
+            const imageUrl = `${this.baseUrl}/job/${jobId}/result/image/${imageName}`;
+            const response = await fetch(imageUrl, { headers });
+            if (!response.ok) {
+              throw new Error(
+                `Failed to download image: ${await response.text()}`,
+              );
+            }
+            const arrayBuffer = await response.arrayBuffer();
+            const buffer = Buffer.from(arrayBuffer);
+            await fs.writeFile(imagePath, buffer);
+
+            images.push(image);
+          }
         }
       }
+      return images;
+    } catch (e) {
+      console.error(`Error while downloading images from the parsed result`, e);
+      if (this.ignoreErrors) {
+        return [];
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * A function to generate an API key.
+   *
+   * @param {string} name - The name for the API key. Must be less than 150 characters.
+   * @param {string} [session] - Optional session information.
+   * @return {Promise<LlamaParseApiResponse>} A promise that resolves to the generated API key response.
+   */
+  async generateApiKey(
```

Comment:
I wouldn't add these wrappers to the LlamaParse API here. 
Better to use https://github.com/fern-api/fern to automatically generate them.

`@llamaindex/cloud` is our package that contains automatically generated API wrappers to the LlamaCloud.

@EmanuelCampos can we add LlamaParse there too?

Reply:
Oh that's cool, in that case I'll leave it up to you, but would be nice to add them there. I'll split out the parts that are not a new wrapper into a new PR and close this one. And sorry if my PR's are a bit meh sometimes.. I'm still learning a lot xD. Thanks for always reviewing!